### DISCUSSION TITLE
Dynamic color text and dark theme

### DISCUSSION THREAD

**Discussion by slawekmoskal** [twbs/bootstrap#38022] [Q&A]
Hi. I have problem with color_contrast settings.

I added a simple mixin to my css file (compile source bootstrap) and a few variables to control the contrast for the hover option for the background.

I also made myself an automatic darkening of the background after hovering, just like the buttons have.

Mixin to contrast text when hovering over a background that can change color.

```
@mixin bg-variant(
   $background,
   $border,
   $color: color-contrast($background),
   $hover-background: if($color == $color-contrast-light, shade-color($background, $btn-hover-bg-shade-amount), tint-color($background, $btn-hover-bg -tint-amount)),
   $hover-border: if($color == $color-contrast-light, shade-color($border, $btn-hover-border-shade-amount), tint-color($border, $btn-hover-border -tint-amount)),
   $hover-color: color-contrast($hover-background),

   --#{$prefix}bg-color: #{$color};
   --#{$prefix}bg-hover-color: #{$hover-color};
   --#{$prefix}bg-hover-bg: #{$hover-background};
   --#{$prefix}bg-hover-border-color: #{$hover-border};
}

```
plus a few variables and everything works great.

I used the buttons shade and tint option - I want the background and buttons to have a similar effect

To the background that I want to change with :hover I added the following variables:

```
.bg-hover {
   &:hover {
     color: var(--#{$prefix}bg-hover-color) !important;
     text-decoration: if($link-hover-decoration == underline, none, null);
     background-color: var(--#{$prefix}bg-hover-bg) !important;
     border-color: var(--#{$prefix}bg-hover-border-color) !important;

     a {
       color: var(--#{$prefix}bg-hover-color) !important;
     }
   }
}

```
And it works without a problem. The problem now is the color of the text but in the data-bs-theme="dark" option.

Then color_contrast doesn't work.

To sum up, I added an option missing in bootstrap - changing the :hover background color, but only for elements that I give the bg-hover class. Because gray-500, gray-600 according to color_contrast should already have different text colors: black and white, I made the dynamic text color change option.

Unfortunately, it does not work in the dark version.

Can this be done with the classes themselves or do I need to add variables to style="" for elements though
 ?

I just want a working option to change the background color for :hover (automatic dim and brighten as for buttons) with color_contrast behavior working in dark theme.

**Comment by julien-deramond** [comment]
IDK if it's possible easily but could you create a StackBlitz (or something else) with a reproducible minimal use case? Would be easier to help you and to suggest a solution/workaround (if there's any).

**Reply by slawekmoskal** [reply]
Hi. I can't run the project despite giving identical data - can you take a look?

https://stackblitz.com/edit/bootstrap-5-mhouu4?file=src%2Findex.html

**Reply by julien-deramond** [reply]
Apparently you call your mixin this way:

```scss
@include bg-variant(
   $hover-background: shade-color($value, $btn-hover-bg-shade-amount),
   $hover-color: shade-color($value, $btn-hover-border-shade-amount)
);
```

But you don't provide anything for the `$background` param. You mixin is `@mixin bg-variant($background, ...)`.

If I add some pre-defined values by doing `@mixin bg-variant($background: #fd0, $border: #df0, ...` the compilation is good. This is not what you want to achieve but it can help you create your minimal example. You can fork mine based on yours if needed: https://stackblitz.com/edit/github-wnkxsq?file=README.md.
